src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,89 @@
+    setViewChecker(true);
+    setSelectedButtonIndex(index);
+  };
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>
             <span className="icon">
-              <i className="far fa-eye" />
+              <i className="fas fa-check" />
             </span>
-          </button>
-        </td>
-      </tr>
+          </th>
+          <th>Title</th>
+          <th> </th>
+        </tr>
+      </thead>
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+      <tbody>
+        {todos.map((todo, index) => (
+          <tr
+            key={todo.id}
+            data-cy="todo"
+            className=""
+          >
+            <td className="is-vcentered">{todo.id}</td>
+            {todo.completed ? (
+              <td className="is-vcentered">
+                <span className="icon" data-cy="iconCompleted">
+                  <i className="fas fa-check" />
+                </span>
+              </td>
+            ) : (
+              <td className="is-vcentered" />
+            )}
+
+            <td className="is-vcentered is-expanded">
+              <p
+                className={classNames({
+                  'has-text-danger': !todo.completed,
+                  'has-text-success': todo.completed,
+                })}
+              >
+                {todo.title}
+              </p>
+            </td>
+            <td className="has-text-right is-vcentered">
+              <button
+                onClick={() => {
+                  handleButtonClick(todo, index);
+                }}
+                data-cy="selectButton"
+                className="button"
+                type="button"
+              >
+                <span className="icon">
+                  <i className={`far ${selectedButtonIndex === index && viewChecker ? 'fa-eye-slash' : 'fa-eye'}`} />
```

Comment:
it looks too complicated.... instead of creating `selectedButtonIndex` and `viewChecker` states

you can just use `selectedTodo`

```
<i className={classNames('far', {
  'fa-eye': selectedTodo !== todo,
  'fa-eye-slash': selectedTodo === todo,
})}
```